src/io.js

```diff
@@ -1,52 +1,61 @@
 import {
   applySpec,
+  ifElse,
   isNil,
   juxt,
   map,
   pipe,
   prop,
   tap,
   unless,
-  when,
 } from "ramda";
 import { asyncExcepts, asyncPairRight, asyncPipe, stack } from "./functional";
 
-export const executeConditionally = (executeQueue, condition) => (
-  clear,
-  resolveAll
-) => when(condition, asyncPipe(tap(clear), executeQueue, resolveAll));
-
-export const batch = (keyFn, waitTime, execute) => {
+/**
+ * Queues the "execute" function until the condition is met or maxWaitTime has passed.
+ * Once one of the above happens we flush the queue and run the execute functions
+ */
+export const batch = (keyFn, maxWaitTime, execute, condition) => {
   const queues = {};
+  const maxWaitTimeout = {};
+
+  const flush_queue = (key) =>
```

Comment:
can we move this outside ?
it can get two functions `get_queue`, `clear_queue`

also maybe rename to `clear_and_execute_queue`

Reply:
You mean outside of the `batch` function?

And the signature would be  `clear_and_execute_queue(get_queue, clear_queue, execute)`?

(just making sure I understand you)

Reply:
yeah exactly. should simplify the code imho